### PR TITLE
S3 fixes

### PR DESCRIPTION
**Pull request by ChGen**
S3 fixes for rpm/deb.
1. Added S3 Storage tests for RPM/DEB/PyPy.
2. Fixed an S3 issue concerning rpm/deb adapters. Test cases were added too.
3. Very old s3mock depends on very old jetty, conflicting with Jetty 12.x in PyPy adapter. So, updating s3mock for PyPy and Jetty revision. 
4. Related issues:
https://github.com/artipie/artipie/issues/1367
https://github.com/artipie/artipie/issues/1368
https://github.com/artipie/artipie/issues/1378
https://github.com/artipie/artipie/issues/1383

As a next step, s3mock and aws sdk will be updated to recent versions for all adapters, since they may affect AWS S3 compatibility and performance too.

### CODE REVIEW

**Comment by ChGen** [comment]
@dgarus , please review.

File: debian-adapter/pom.xml [hunk 1]
```diff
@@ -93,6 +93,54 @@ SOFTWARE.
...
+    </dependency>
+
+    <!-- s3 mocks deps -->
+    <dependency>
+      <groupId>com.adobe.testing</groupId>
+      <artifactId>s3mock</artifactId>
+      <version>2.4.9</version>
+      <scope>test</scope>
+      <!--<exclusions>
+        <exclusion>
```
Comment (dgarus) [inline debian-adapter/pom.xml:110]: Do we really need these commented dependencies here?

File: asto/asto-core/src/main/java/com/artipie/asto/rx/RxStorageWrapper.java [hunk 2]
```diff
@@ -26,53 +29,73 @@ public final class RxStorageWrapper implements RxStorage {
...
     }
 
     @Override
     public Single<Content> value(final Key key) {
-        return Single.defer(() -> SingleInterop.fromFuture(this.storage.value(key)));
+        return Single.defer(() -> SingleInterop.fromFuture(
+            this.storage.value(key).thenCompose(content -> {
+                return CompletableFuture.completedFuture(content);
+            })
+        )).observeOn(this.scheduler);
```
Comment (dgarus) [inline asto/asto-core/src/main/java/com/artipie/asto/rx/RxStorageWrapper.java:93]:
```suggestion
        return Single.defer(() -> SingleInterop.fromFuture(
            this.storage.value(key).thenCompose(CompletableFuture::completedFuture)
        )).observeOn(this.scheduler);
```

File: http-client/pom.xml [hunk 3]
```diff
@@ -34,7 +34,7 @@ SOFTWARE.
   <name>Artipie HTTP client</name>
   <url>https://github.com/artipie/http-client</url>
   <properties>
-    <jettyVersion>12.0.3</jettyVersion>
+    <jettyVersion>12.0.6</jettyVersion>
```
Comment (dgarus) [inline http-client/pom.xml:37]: `jettyVersion` property should be defined in the parent pom, isn't it?

**Review by dgarus — CHANGES_REQUESTED** [review]
Thanks.
Please check my comments

**Comment by ChGen** [comment]
@dgarus , please review.

**Review by dgarus — APPROVED** [review]

**Pull request merged**